### ISSUE TITLE
Feature request: skip files for a specific linter.

### ISSUE THREAD

**Issue by misha-tectonic** [golangci/golangci-lint#320]
It would be useful to have an option to skip files for a specific linter.

A use-case: do not run `unparam` for `*_test.go`, as stub functions defined in tests are triggering false positives.

Version: v1.12.2

`.golangci.yml`:
```
run:
  deadline: 2m

issues:
  max-issues-per-linter: 0
  max-same-issues: 0
  exclude:
    # This project does not have a rule to avoid unkeyed fields
    - composite literal uses unkeyed fields
    # This project does not require all parameters to be used
    - "`.*` - `.*` is unused"

linters:
  enable-all: true
  disable:
    # This project does not have a rule to have no globals
    - gochecknoglobals
    # This project does not have a rule to have no inits
    - gochecknoinits
    # This test has too many false positives
    - gocyclo
    # This project does not have a standard line length
    - lll
    # This project does not have a rule to forbid naked returns
    - nakedret
    # This project does not care about optimizing out few bytes of memory
    - maligned
```

**Comment by jirfag** [comment]
Hi! I agree, it's one of the most useful feature we need to implement

**Comment by ernado** [comment]
This issue blocks me from transferring from gometalinter.
Part of my "configuration":
```
-e "_test.go.+(gocyclo|errcheck|dupl)" \
-e "Message.+\(aligncheck\)" \
-e "e2e\/.+(gocyclo|errcheck|dupl)" \
-e "internal\/hmac\/hmac_test.+(lll)" \
-e "internal\/hmac\/.+Errors unhandled" \
```

Actually the ability to regex-match the full error (including the file path and linter name) would be really great.
E.g. when excluding, use the full string
```
internal/hmac/pool_test.go:71:1: cyclomatic complexity 11 of func `TestHMACPool_SHA1` is high (> 10) (gocyclo)
```
Instead of just ```cyclomatic complexity 11 of func `TestHMACPool_SHA1` is high (> 10)```

**Comment by ernado** [comment]
@jirfag @misha-tectonic 
What do you think about the following syntax?
I've already implemented `exclude-rules` processing locally and can 
start preparing a PR if you are OK with it.
```yaml
issues:
  exclude-rules:
    # Exclude some linters from running on tests files.
    - path: _test\.go
      linters:
        - gocyclo
        - errcheck
        - dupl
        - gosec

    # Ease some gocritic warnings on test files.
    - path: _test\.go
      text: "(unnamedResult|exitAfterDefer)"
      linters:
        - gocritic

    # Exclude known linters from partially hard-vendored code,
    # which is impossible to exclude via "nolint" comments.
    - path: internal/hmac/
      text: "weak cryptographic primitive"
      linters:
        - gosec
    - path: internal/hmac/
      text: "Write\\` is not checked"
      linters:
        - errcheck

    # Ease linting on benchmarking code.
    - path: cmd/stun-bench/
      linters:
        - gosec
        - errcheck
  enable-all: true
  disable:
    - prealloc
    - gochecknoglobals
```

Technically we can have something like
```yml
  exclude:
    - "foo"
    - "bar"
    - path: _test\.go
      linters:
        - gocyclo
        - errcheck
        - dupl
        - gosec
```
But it will sufficiently complicate configuration parsing.

**Comment by jirfag** [comment]
I've merged @ernado fix, thank you!
